src/MongoDB.Driver.Core/Core/Operations/AsyncCursor.cs

```diff
@@ -273,39 +273,55 @@ private BsonDocument CreateKillCursorsCommand()
         private CursorBatch<TDocument> ExecuteGetMoreCommand(IChannelHandle channel, CancellationToken cancellationToken)
         {
             var command = CreateGetMoreCommand();
-            var result = channel.Command<BsonDocument>(
-                _channelSource.Session,
-                null, // readPreference
-                _collectionNamespace.DatabaseNamespace,
-                command,
-                null, // commandPayloads
-                NoOpElementNameValidator.Instance,
-                null, // additionalOptions
-                null, // postWriteAction
-                CommandResponseHandling.Return,
-                __getMoreCommandResultSerializer,
-                _messageEncoderSettings,
-                cancellationToken);
+            BsonDocument result;
+            try
+            {
+                result = channel.Command<BsonDocument>(
+                    _channelSource.Session,
+                    null, // readPreference
+                    _collectionNamespace.DatabaseNamespace,
+                    command,
+                    null, // commandPayloads
+                    NoOpElementNameValidator.Instance,
+                    null, // additionalOptions
+                    null, // postWriteAction
+                    CommandResponseHandling.Return,
+                    __getMoreCommandResultSerializer,
+                    _messageEncoderSettings,
+                    cancellationToken);
+            }
+            catch (MongoCommandException ex) when (ex.Code == (int)ServerErrorCode.CursorNotFound)
```

Comment:
Consider adding this into a helper method

Reply:
Done.